### ISSUE TITLE
GraphML: inconsistent handling of empty attribute value

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#2057]
**Describe the bug**

In GraphML, empty numeric attributes values such as `<data></data>` are interpreted as NaN. However, when there is whitespace, such as `<data> </data>`, it is interpreted as 0. This is inconsistent.

**To reproduce**

```
<?xml version="1.0" encoding="UTF-8"?>
<graphml xmlns="http://graphml.graphdrawing.org/xmlns"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://graphml.graphdrawing.org/xmlns
         http://graphml.graphdrawing.org/xmlns/1.0/graphml.xsd">
<!-- Created by igraph -->
  <key id="v_weight" for="node" attr.name="weight" attr.type="float"/>
  <graph id="G" edgedefault="undirected">
    <node id="n0">
      <data key="v_weight">1</data>
    </node>
    <node id="n1">
      <data key="v_weight"> </data>
    </node>
    <node id="n2">
      <data key="v_weight"></data>
    </node>
    <node id="n3">
      <data key="v_weight"></data>
    </node>
    <edge source="n0" target="n1">
    </edge>
    <edge source="n1" target="n2">
    </edge>
    <edge source="n1" target="n3">
    </edge>
    <edge source="n2" target="n3">
    </edge>
  </graph>
</graphml>
```

Test program:

```

#include <igraph.h>
#include <assert.h>

int main(int argc, char *argv[]) {
    igraph_t graph;
    FILE *file;

    assert(argc == 2);
    file = fopen(argv[1], "r");
    assert(file);

    igraph_set_attribute_table(&igraph_cattribute_table);

    igraph_set_error_handler(&igraph_error_handler_printignore);
    if (igraph_read_graph_graphml(&graph, file, 0) == IGRAPH_SUCCESS) {
        igraph_write_graph_gml(&graph, stdout, NULL, NULL);
        igraph_destroy(&graph);
    }
    fclose(file);

    return 0;
}
```

**Version information**
`develop`

**Comment by szhorvat** [comment]
There are multiple issues here. One is that `igraph_i_parse_real()` returns 0 for empty strings. It should _probably_ made to raise an error instead.  The whitespace case, i.e. `<data> </data>` is handled in `igraph_i_graphml_parse_numeric()`, which is passed a default value of 0 for some reason, not sure why.

I'm not sure where exactly the handling of the `<data></data>` happens, which produces NaN.

**Comment by ntamas** [comment]
`igraph_i_parse_*()` functions are fixed in bfed9a1b5 ; they now return an error for empty strings. The GraphML parsing is fixed in e8832c977
